tests/std/tests/P0896R4_ranges_alg_find_first_of/test.cpp

```diff
@@ -2,92 +2,86 @@
+            auto result = find_first_of(wrapped_haystack.begin(), wrapped_haystack.end(), wrapped_needle.begin(),
+                wrapped_needle.end(), pred, get_first);
+            STATIC_ASSERT(same_as<decltype(result), iterator_t<Read1>>);
+            assert(result.peek() == begin(haystack) + 2);
+        }
 
-            BinaryPredicateFor<iterator_t<In1>, iterator_t<Fwd2>> pred{};
-            (void) ranges::find_first_of(in1, fwd2, pred);
-            (void) ranges::find_first_of(
-                ranges::begin(in1), ranges::end(in1), ranges::begin(fwd2), ranges::end(fwd2), pred);
+        { // Validate range overload [not found case]
+            Read1 wrapped_haystack{haystack};
+            Read2 wrapped_needle{bad_needle};
 
-            HalfProjectedBinaryPredicateFor<iterator_t<Fwd2>> halfpred{};
-            ProjectionFor<iterator_t<In1>> halfproj{};
-            (void) ranges::find_first_of(in1, fwd2, halfpred, halfproj);
-            (void) ranges::find_first_of(
-                ranges::begin(in1), ranges::end(in1), ranges::begin(fwd2), ranges::end(fwd2), halfpred, halfproj);
+            auto result = find_first_of(wrapped_haystack, wrapped_needle, pred, get_first);
+            STATIC_ASSERT(same_as<decltype(result), iterator_t<Read1>>);
+            assert(result.peek() == end(haystack));
+        }
+        {
+            // Validate iterator + sentinel overload [not found case]
+            Read1 wrapped_haystack{haystack};
+            Read2 wrapped_needle{bad_needle};
 
-            ProjectedBinaryPredicate<0, 1> projpred{};
-            ProjectionFor<iterator_t<In1>, 0> proj1{};
-            ProjectionFor<iterator_t<Fwd2>, 1> proj2{};
-            (void) ranges::find_first_of(in1, fwd2, projpred, proj1, proj2);
-            (void) ranges::find_first_of(
-                ranges::begin(in1), ranges::end(in1), ranges::begin(fwd2), ranges::end(fwd2), projpred, proj1, proj2);
+            auto result = find_first_of(wrapped_haystack.begin(), wrapped_haystack.end(), wrapped_needle.begin(),
+                wrapped_needle.end(), pred, get_first);
+            STATIC_ASSERT(same_as<decltype(result), iterator_t<Read1>>);
+            assert(result.peek() == end(haystack));
         }
     }
 };
 
-template void test_in_fwd<instantiator, const int, const int>();
+#ifdef TEST_EVERYTHING
+int main() {
+    STATIC_ASSERT((test_in_fwd<instantiator, const P, const int>(), true));
+    test_in_fwd<instantiator, const P, const int>();
+}
+#else // ^^^ test all range combinations // test only interesting range combos vvv
+using in_test_range  = test::range<input_iterator_tag, const P, test::Sized::no, test::CanDifference::no,
+    test::Common::no, test::CanCompare::no, test::ProxyRef::yes>;
+using fwd_test_range = test::range<forward_iterator_tag, const int, test::Sized::no, test::CanDifference::no,
+    test::Common::no, test::CanCompare::yes, test::ProxyRef::yes>;

```

Comment:
These can be moved into `run_tests`, ideally with some justification for why they provide sufficient coverage like "The implementation of algorithm is oblivious to any guarantees beyond its minimum requirements, so no additional coverage is necessary." (I _hate_ that the correctness of the test depends on someone paying enough attention to realize they may need to update the test when they modify the algorithm, but I hope being noisy about it will help.)

Reply:
Added a few words